applicationset/utils/utils.go

```diff
@@ -21,6 +21,16 @@ import (
 	argoappsetv1 "github.com/argoproj/argo-cd/v2/pkg/apis/applicationset/v1alpha1"
 )
 
+var sprigFuncMap = sprig.GenericFuncMap() // a singleton for better performance
+
+func init() {
+	// Avoid allowing the user to learn things about the environment.
+	delete(sprigFuncMap, "env")
```

Comment:
I would add:

```
delete(funcMap, "expandenv")
delete(funcMap, "env")
delete(funcMap, "getHostByName")
delete(funcMap, "base")
delete(funcMap, "dir")
delete(funcMap, "clean")
delete(funcMap, "ext")
delete(funcMap, "isAbs")
delete(funcMap, "osBase")
delete(funcMap, "osClean")
delete(funcMap, "osDir")
delete(funcMap, "osExt")
delete(funcMap, "osIsAbs")
```

to delete all theses:

```
// OS:
	"env":       os.Getenv,
	"expandenv": os.ExpandEnv,

	// Network:
	"getHostByName": getHostByName,

	// Paths:
	"base":  path.Base,
	"dir":   path.Dir,
	"clean": path.Clean,
	"ext":   path.Ext,
	"isAbs": path.IsAbs,

	// Filepaths:
	"osBase":  filepath.Base,
	"osClean": filepath.Clean,
	"osDir":   filepath.Dir,
	"osExt":   filepath.Ext,
	"osIsAbs": filepath.IsAbs,
```

Reply:
I took another look, and I think those are safe. They don't tell you anything about the filesystem besides which path separator the OS uses. All the operations are just string operations.

Reply:
Perfect